src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -223,7 +223,7 @@ public TElement EnqueueDequeue(TElement element, TPriority priority)
         {
             var root = _nodes[RootIndex];
 
-            if (Comparer.Compare(priority, root.Priority) < 0)
+            if (Comparer.Compare(priority, root.Priority) <= 0)
```

Comment:
Do we have tests that check for duplicate elements that might have failed before this fix was made?

Reply:
Covered in `PriorityQueue_Generic_EnqueueDequeue_EqualToMin`.